### PR TITLE
ci: Add Github action workflow ci_win.yml file

### PR DESCRIPTION
**Pull request by mrobertseidowsky-gpsw**
- This PR introduces the CI file for build on Windows (here windows-latest but can be fixed to windows-2019).
- I tested the file on my forked repo, see [build](https://github.com/mrobertseidowsky-gpsw/gopro-lib-node.gl/runs/1298599290?check_suite_focus=true)
- Installation of dependencies takes around 10min and the build takes around 1min.
- The env variable CHERE_INVOKING is necessary to stay in the same repository, if not, the Makefile is not found.
- Don't forget to update your README with a new badge !

### CODE REVIEW

File: .github/workflows/ci_win.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      - name: Install dependencies
+        run: |
+          C:\msys64\usr\bin\bash -lc "pacman --noconfirm -Syuu"
+          C:\msys64\usr\bin\bash -lc "pacman --noconfirm -S git make"
+          C:\msys64\usr\bin\bash -lc "pacman -S --noconfirm mingw-w64-x86_64-{toolchain,ffmpeg,python}"
+
+      - name: Build
+        run: |
+          $env:CHERE_INVOKING = 'yes'  # Preserve the current working directory
+          C:\msys64\usr\bin\bash -lc "make TARGET_OS=MinGW-w64"
```
Comment (mbouron-gpsw) [inline .github/workflows/ci_win.yml:31]: Would it make sense to add -jN to speedup the build process if that's possible ?

**Comment by mbouron-gpsw** [comment]
Other than my remark, LGTM, thanks.

File: .github/workflows/ci_win.yml (hunk 1 above)
Comment (mrobertseidowsky-gpsw) [reply .github/workflows/ci_win.yml]: yes it's possible, it's added.

File: .github/workflows/ci_win.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      - uses: msys2/setup-msys2@v2
+        with:
+          msystem: MINGW64 # Start a 64 bit Mingw environment
+          update: true
+
+      - name: Install dependencies
+        run: |
+          C:\msys64\usr\bin\bash -lc "pacman --noconfirm -Syuu"
+          C:\msys64\usr\bin\bash -lc "pacman --noconfirm -S git make"
+          C:\msys64\usr\bin\bash -lc "pacman -S --noconfirm mingw-w64-x86_64-{toolchain,ffmpeg,python}"
```
Comment (cboesch-gpsw) [inline .github/workflows/ci_win.yml:26]: You may want to add `--as-needed` so it's not re-installed if already present

File: .github/workflows/ci_win.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      - uses: actions/checkout@v2
+
+      - uses: msys2/setup-msys2@v2
+        with:
+          msystem: MINGW64 # Start a 64 bit Mingw environment
+          update: true
+
+      - name: Install dependencies
+        run: |
+          C:\msys64\usr\bin\bash -lc "pacman --noconfirm -Syuu"
```
Comment (cboesch-gpsw) [inline .github/workflows/ci_win.yml:24]: > * Installation of dependencies takes around 10min and the build takes around 1min.

Can you save some time by not running the `-Syuu` and trusting the image to do the correct thing?

Comment (mrobertseidowsky-gpsw) [reply .github/workflows/ci_win.yml]: Actually, it works without this line! and the "Install dependencies" step takes only around 4min now. Line removed.

File: .github/workflows/ci_win.yml (hunk 2 above)
Comment (mrobertseidowsky-gpsw) [reply .github/workflows/ci_win.yml]: Option "--needed" added, thanks.

**Comment by mrobertseidowsky-gpsw** [comment]
I also added the new badge in this PR.

**Pull request merged**